### PR TITLE
Bug 1322587 - NotFound error (404) is logged but doesn't terminate pruning

### PR DESCRIPTION
**Pull request by soltysh**
This unblocks the problem described in [this bug](https://bugzilla.redhat.com/show_bug.cgi?id=1322587). 

@miminar @pweil- ptal
@wshearn fyi

### CODE REVIEW

**Comment by pweil-** [comment]
LGTM

@smarterclayton FYI

File: pkg/image/prune/imagepruner.go [hunk 1]
```diff
@@ -916,6 +916,12 @@ func deleteFromRegistry(registryClient *http.Client, url string) error {
 		}
 		defer resp.Body.Close()
 
+		// TODO: investigate why we're getting non-existent layers, for now we're logging
+		// them out and continue working
+		if resp.StatusCode == http.StatusNotFound {
+			glog.Infof("Got %v for %s", resp.Status, url)
```
Comment (smarterclayton) [inline pkg/image/prune/imagepruner.go]: glog.Warning, better message like `Unable to prune layer: %s returned 404`

**Comment by smarterclayton** [comment]
One comment

**Comment by soltysh** [comment]
@smarterclayton comment addressed.

**Comment by smarterclayton** [comment]
Approved LGTM [merge]

**Pull request merged**